src/ParallelAlgorithms/LinearSolver/Schwarz/ElementCenteredSubdomainData.hpp

```diff
@@ -13,9 +13,15 @@
 #include "NumericalAlgorithms/LinearSolver/InnerProduct.hpp"
 #include "ParallelAlgorithms/LinearSolver/Schwarz/OverlapHelpers.hpp"
 #include "Utilities/MakeWithValue.hpp"
+#include "Utilities/TMPL.hpp"
 
 namespace LinearSolver::Schwarz {
 
+/// \cond
+template <bool Const, size_t Dim, typename TagsList>
+struct ElementCenteredSubdomainDataIterator;
```

Comment:
interesting pattern, though I don't quite understand how the data access would actually work in that case. Does the subclass implement some sort of non-owning access to the superclass? In any case, I think the current design works and doesn't need to be re-written.

Reply:
@moxcodes I generally like that pattern, but it often comes into conflict with our rule that `operator==` should be a free function, because template argument deduction only considers the class itself, not any superclasses.  (I.e., `template <typename X> void f(A<X>::B)` can't deduce X.)